packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -617,6 +630,12 @@ export class DbRemoteDocument {
 
   static collectionReadTimeIndexPath = ['parentPath', 'readTime'];
 
+  // TODO: We are currently storing full document keys almost three times
+  // (once as part of the primary key, once - partly - as `parentPath` and once
+  // inside the encoded documents). During our next migration, we should
```

Comment:
We store it 3 more times if you include the indexes (collectionReadTimeIndex stores it twice, readTimeIndex stores it once).  Hopefully people don't use deeply-nested subcollections and small documents!

It occurs to me that we could maintain our own collectionReadTimeIndex that instead of being a true index was just an object store containing `{collectionPath, readTime, documentId}` objects.  This would save us storing the full documentKey.  It's probably not worth doing for web since it'd be less efficient than a real index, but with LevelDB (where we have to implement our own indexes anyway), it probably makes sense.